bots/image-prepare

```diff
@@ -46,100 +47,141 @@ def main():
+        if build_image and not os.path.exists(build_image):
             subprocess.check_call([ "image-download", build_image])
-        if install_image:
-            subprocess.check_call([ "image-download", install_image])
     except OSError, ex:
         if ex.errno != errno.ENOENT:
             raise
         sys.stderr.write("image-prepare: missing tools to download images\n")
     except subprocess.CalledProcessError as e:
-        print "unable to download all necessary images", e
+        sys.stderr.write("image-prepare: unable to download all necessary images\n")
         return 1
 
     try:
-        build_and_install(install_image, build_image, results, args={
-            "verbose": args.verbose,
-            "sit": args.sit,
-            "quick": args.quick,
-            "build_image": build_image,
-            "build_only": args.build_only,
-            "install_only": args.install_only,
-            "containers": args.containers,
-            "address": args.address,
-        })
-    except RuntimeError as ex:
+        if not args.install_only:
+            build_and_install(build_image, results, skips, args)
+        if not args.build_only and build_image != base_image:
+            only_install(base_image, results, skips, args, args.address)
+    except (RuntimeError, testvm.Failure) as ex:
         sys.stderr.write("image-prepare: {0}\n".format(str(ex)))
         return 2
 
+    # The remaining images not yet prepared are symlinked from the bots/images
+    # directory into the test/images directory. This allows for other branches to
+    # run without modification, and allows tests to load images that they didn't
+    # install or need to install cockpit on (above).
+    link_remaining_images()
+
     return 0
 
+def link_remaining_images():
+    bots_images = os.path.join(BOTS, "images")
+    test_images = os.path.join(TEST, "images")
+    if not os.path.exists(test_images):
+        os.makedirs(test_images)
+    for image in os.listdir(bots_images):
+        if "." in image:
+            continue
+        if os.path.isdir(os.path.join(bots_images, image)):
+            continue
+        dest = os.path.join(test_images, image)
+        if not os.path.isfile(dest):
+            if os.path.lexists(dest):
+                os.unlink(dest)
+            os.symlink(os.path.join("..", "..", "bots", "images", image), dest)
+
 def upload_scripts(machine, args):
     machine.execute("rm -rf /var/lib/testvm")
     machine.upload([ os.path.join(BOTS, "images", "scripts", "lib") ], "/var/lib/testvm")
     machine.upload([ os.path.join(BOTS, "images", "scripts", "%s.install" % machine.image) ], "/var/tmp")
     machine.upload([ os.path.join(BASE, "containers") ], "/var/tmp")
 
+# Create the necessary layered image for the build/install
+def prepare_backing_file(machine, image):
+    if "/" in image:
+        image = os.path.abspath(image)
+    image = os.path.join(BOTS, "images", image)
```

Comment:
`os.path.join()` does ignore prefixes when it encounters an absolute path component, but this is not entirely obvious. It would be easier to grasp if you used `else:` here.